File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-19.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> seen = new HashSet<>();
        int max =0;
        int j=0;
        for(int i=0;i<s.length();i++){
            while(seen.contains(s.charAt(i))){
                seen.remove(s.charAt(j));
                j++;
            }
            seen.add(s.charAt(i));
            max = Math.max(max,i-j+1);
        }
        return max;
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-19.py
class Solution:
    def isValid(self, s: str) -> bool:
        map = {'{':'}','[':']','(':')'} 
        seen = []
        for i in s:
            if i == '{' or i == '(' or i == '[':
                seen.append(i)
            else:
                if not seen or i != map[seen[-1]]:
                    return False
                else:
                    seen.pop()
        if seen:
            return False
        else:
            return True